testing/dart/image_shader_test.dart

```diff
@@ -29,5 +45,15 @@ void main() {
     final Paint paint = Paint()..shader=shader;
     const Rect rect = Rect.fromLTRB(0, 0, 100, 100);
     testCanvas((Canvas canvas) => canvas.drawRect(rect, paint));
+
+    if (assertsEnabled) {
+      expect(shader.debugDisposed, false);
+    }
+    shader.dispose();
+    if (assertsEnabled) {
+      expect(shader.debugDisposed, true);
+    }
+
+    image.dispose();
```

Comment:
A test that tries to attach a disposed shader to a `Paint` and then use the `Paint` could be useful. It should not segfault or abort, and should rather throw a Dart exception.